includes/SetupWizard.php

```diff
@@ -316,13 +313,16 @@ public function save_step() {
 					update_option( $option, $new_settings );
 				}
 			} elseif ( ! empty( $request['wpo_wcpdf_step'] ) && 'show-action-buttons' === $request['wpo_wcpdf_step'] ) {
+				$orders_column_hidden_key = \Automattic\WooCommerce\Utilities\OrderUtil::custom_orders_table_usage_is_enabled() ? 'managewoocommerce_page_wc-orderscolumnshidden' : 'manageedit-shop_ordercolumnshidden';
+				$user_id                  = get_current_user_id();
+				$hidden                   = get_user_meta( $user_id, $orders_column_hidden_key, true );
 				if ( ! empty( $request['wc_show_action_buttons'] ) ) {
 					$hidden = array_filter( $hidden, function( $setting ){ return $setting !== 'wc_actions'; } );
-					update_user_meta( $user_id, 'manageedit-shop_ordercolumnshidden', $hidden );
 				} else {
 					array_push( $hidden, 'wc_actions' );
```

Comment:
Maybe just `$hidden[] = 'wc_actions';` would be enough here? (not your code fault)

```php
if ( ! in_array( 'wc_actions', $hidden, true ) ) {
	$hidden[] = 'wc_actions';
}
```